lib/v2/screens/dashboard/components/transactions_list/transactions_list_widget.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/datasource/local/settings_storage.dart';
+import 'package:seeds/v2/datasource/remote/model/transaction_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/dashboard/components/transactions_list/interactor/viewmodels/transactions_list_bloc.dart';
+import 'package:seeds/v2/screens/dashboard/components/transactions_list/interactor/viewmodels/transactions_list_events.dart';
+import 'package:seeds/v2/screens/dashboard/components/transactions_list/interactor/viewmodels/transactions_list_state.dart';
+import 'package:seeds/v2/screens/dashboard/components/transactions_list/transaction_info_card.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/bloc.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/wallet_bloc.dart';
+import 'package:shimmer/shimmer.dart';
+
+class TransactionsListWidget extends StatefulWidget {
+  @override
+  _TransactionsListWidgetState createState() => _TransactionsListWidgetState();
+}
+
+class _TransactionsListWidgetState extends State<TransactionsListWidget> with AutomaticKeepAliveClientMixin {
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return BlocProvider<TransactionsListBloc>(
+      create: (_) => TransactionsListBloc()..add(LoadTransactionsListEvent()),
+      child: BlocListener<WalletBloc, WalletState>(
+          listenWhen: (context, state) => state.pageState == PageState.loading,
+          listener: (context, state) {
+            BlocProvider.of<TransactionsListBloc>(context).add(LoadTransactionsListEvent());
+          },
```

Comment:
Unused callbacks. 
Also It is important to note that `listenWhen` it doesn't have a `context`  and `state` callbacks, it has 2 states callbacks (`previous` and `current`).

```
          listenWhen: (_, current) => current.pageState == PageState.loading,
          listener: (context, _) {
            BlocProvider.of<TransactionsListBloc>(context).add(LoadTransactionsListEvent());
          },
```

Reply:
Thanks that's a bug